--- /usr/src/dahdi-2.5.0.1+dfsg-1ubuntu3/drivers/dahdi/xpp/xbus-sysfs.c.ori	2014-06-04 23:18:24.644291060 +0200
+++ /usr/src/dahdi-2.5.0.1+dfsg-1ubuntu3/drivers/dahdi/xpp/xbus-sysfs.c	2014-06-04 23:22:30.945233860 +0200
@@ -58,10 +58,19 @@
 	return exec_sync_command(buf, count);
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 13, 0)
 static struct driver_attribute xpp_attrs[] = {
 	__ATTR(sync, S_IRUGO | S_IWUSR, sync_show, sync_store),
 	__ATTR_NULL,
 };
+#else
+static DRIVER_ATTR_RW(sync);
+static struct attribute *xpp_attrs[] = {
+	&driver_attr_sync.attr,
+	NULL,
+};
+ATTRIBUTE_GROUPS(xpp);
+#endif
 
 /*--------- Sysfs Bus handling ----*/
 static DEVICE_ATTR_READER(xbus_state_show, dev, buf)
@@ -457,7 +466,11 @@
 	.uevent         = astribank_uevent,
 #endif
 	.dev_attrs	= xbus_dev_attrs,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 13, 0)
 	.drv_attrs	= xpp_attrs,
+#else
+	.drv_groups	= xpp_groups,
+#endif
 };
 
 static int astribank_probe(struct device *dev)
